src/grpc/packet_router/hpr_packet_router_downlink_handler.erl

```diff
@@ -0,0 +1,45 @@
+-module(hpr_packet_router_downlink_handler).
+
+-behaviour(grpcbox_client_stream).
+
+-export([
+    init/3,
+    handle_message/2,
+    handle_headers/2,
+    handle_trailers/4,
+    handle_eos/1
+]).
+
+-type stream_id() :: non_neg_integer().
+-type callback_data() :: #{gateway => libp2p_crypto:pubkey_bin(), stream_id => stream_id()}.
```

Comment:
This has to be a `map()` right? We can't make this a record?

Reply:
Yes, we can.